web/src/compose.js

```diff
@@ -144,13 +144,11 @@ export function clear_compose_box() {
 
 export function send_message_success(request, data) {
     if (!request.locally_echoed) {
-        if ($("textarea#compose-textarea").data("draft-id")) {
-            drafts.draft_model.deleteDraft($("textarea#compose-textarea").data("draft-id"));
-        }
         clear_compose_box();
     }
 
     echo.reify_message_id(request.local_id, data.id);
+    drafts.draft_model.deleteDraft(request.draft_id);
```

Comment:
Does this work is the `request.draft_id` is undefined? Is that possible?

Reply:
Yes it does work (silently does nothing) when `draft_id` is undefined, and that case is possible whenever the message is under 3 characters when sending (a realistic case would be "ok"). 

If this sent message was ever drafted earlier (which means it was once draftable = 3 characters or over) that version will remain in drafts as expected. And if this 1/2 character message sent is new, that means it won't be drafted at all.